pkg/flags/urls.go

```diff
@@ -64,17 +77,35 @@ func NewURLsValue(init string) (*URLsValue, error) {
 	return v, err
 }
 
-// ParseHostPortAddr returns a host:port list
+// ParseHostPortAddr returns a scheme://host:port or host:port list
 func ParseHostPortAddr(s string) ([]string, error) {
 	strs := strings.Split(s, ",")
 	addrs := make([]string, 0, len(strs))
+
 	for _, str := range strs {
 		str = strings.TrimSpace(str)
-		_, _, err := net.SplitHostPort(str)
+
+		// str may looks like 127.0.0.1:8000
+		if _, _, err := net.SplitHostPort(str); err == nil {
+			addrs = append(addrs, str)
+			continue
+		}
+
+		u, err := url.Parse(str)
 		if err != nil {
-			return nil, errors.Trace(err)
+			return nil, errors.Errorf("parse url %s failed %v", str, err)
 		}
-		addrs = append(addrs, str)
+		if u.Scheme != "http" && u.Scheme != "https" && u.Scheme != "unix" && u.Scheme != "unixs" {
+			return nil, errors.Errorf("URL scheme must be http, https, unix, or unixs: %s", str)
+		}
+		if _, _, err := net.SplitHostPort(u.Host); err != nil {
+			return nil, errors.Errorf(`URL address does not have the form "host:port": %s`, str)
+		}
```

Comment:
unix??

Even if you're going to support unix-domain sockets, the `unix:///tmp/etcd.sock` will have an empty `u.Host` and cause `net.SplitHostPort` to fail ("missing port in address").

Reply:
I don'ts what it wants actually, but after change changing to using `flags.NewURLsValue`
the endpoings get here will be int the formort:  `unix://localhost:4085651753091690150`
https://github.com/pingcap/tidb-binlog/blob/649ec60f9ceb65fbbaf22e06e43e39ae62e56df9/pump/node_test.go#L58

there's some similar code it wil use by `flag.NewUrLsValue`...
https://github.com/pingcap/tidb-binlog/blob/649ec60f9ceb65fbbaf22e06e43e39ae62e56df9/pkg/types/urls.go#L40